packet.go

```diff
@@ -52,26 +69,58 @@ func unmarshalUint32(b []byte) (uint32, []byte) {
 	return v, b[4:]
 }
 
+func unmarshalUint32Safe(b []byte) (uint32, []byte, error) {
```

Comment:
please make this wrap unmarshalUint32, do the same for the other *safe derivatives.